moodler/download.py

```diff
@@ -92,9 +95,11 @@ def download_all_submissions(assignment_id, assignment_name, output_path, sessio
     """
     # Build the get request.
     params = {"id": assignment_id, "action": "downloadall"}
-    response = session.get(URL + "/mod/assign/view.php", params=params)
-
-    # TODO: Raise an exception in case the file download failed
+    try:
+        response = session.get(URL + "/mod/assign/view.php", params=params)
+    except ConnectionError as exc:
+        logger.exception(exc)
```

Comment:
`logger.exception` automatically logs the exception, no need to give it `exc`
Instead give it a meaningful message like in the exception you raise